core/src/main/scala/kafka/coordinator/GroupMetadata.scala

```diff
@@ -282,6 +282,12 @@ private[coordinator] class GroupMetadata(val groupId: String, initialState: Grou
     pendingOffsetCommits ++= offsets
   }
 
+  def removeOffsets(topicPartition: TopicPartition) = {
+    val removedOffsets = offsets.filter( _._1 == new TopicPartition(topicPartition.topic, topicPartition.partition))
```

Comment:
Yes, it bit tricky top clean pendingOffsetCommits. I implemented some changes..Pl review..